core/domain/collection_services_test.py

```diff
@@ -739,6 +741,17 @@ def test_add_node(self):
         self.assertEqual(
             collection.exploration_ids, [self.EXPLORATION_ID, _NEW_EXP_ID])
 
+    def test_add_node_with_invalid_exploration(self):
+        _INVALID_EXP_ID = 'invalid_exploration_id'
+        collection = collection_services.get_collection_by_id(
+            self.COLLECTION_ID)
+        with self.assertRaises(Exception):
```

Comment:
Please use ``self.assertRaisesRegexp(error, substring)`` instead so that you can also match part of the error string and ensure the right error is being thrown for this behavior.